client/src/components/qa/QuestionList.jsx

```diff
@@ -1,19 +1,45 @@
-import React from 'react';
+/* eslint-disable no-unused-vars */
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
 import Question from './Question.jsx';
+import AddQuestion from './AddQuestion.jsx';
 
-function QuestionList({ qList }) {
+function QuestionList({ questionList, getQuestions }) {
+  const [questionCount, setQuestionCount] = useState(2);
+
+  if (questionCount >= questionList.length) {
+    return (
+      <div id="QuestionList">
+        {questionList.slice(0, questionCount).map((question) => (
+          <Question
+            body={question.question_body}
+            helpfulness={question.question_helpfulness}
+            id={question.question_id}
+            key={question.question_id}
+            answers={question.answers}
+            getQuestions={getQuestions}
+          />
+        ))}
+      </div>
+    );
+  }
   return (
-    <div>
-      -----------------List of questions-----------------
-      {qList.map((question) => (
+    <div id="QuestionList">
+      {questionList.slice(0, questionCount).map((question) => (
         <Question
           body={question.question_body}
+          helpfulness={question.question_helpfulness}
+          id={question.question_id}
           key={question.question_id}
           answers={question.answers}
+          getQuestions={getQuestions}
         />
       ))}
+      <button onClick={(e) => { setQuestionCount(questionCount + 2); }}>
```

Comment:
Same idea here with the conditional on line 10, you can get rid of lines 23-37 and place the reverse condition of `questionCount < questionList.length &&` followed by current code on lines 38-41 to not repeat some of these lines of code if you want.